src/gui/attributetable/qgsfieldconditionalformatwidget.cpp

```diff
@@ -0,0 +1,209 @@
+  editStyle( index.row(), style );
+}
+
+void QgsFieldConditionalFormatWidget::editStyle( int editIndex, QgsConditionalStyle style )
+{
+  pages->setCurrentIndex( 1 );
+  mEditIndex = editIndex;
+  mEditing = true;
+
+  mRuleEdit->setText( style.rule() );
+  mDeleteButton->show();
+  btnBackgroundColor->setColor( style.backgroundColor() );
+  btnTextColor->setColor( style.textColor() );
+  if ( !style.icon().isNull() )
+  {
+    checkIcon->setChecked( true );
+    QIcon icon( style.icon() );
+    btnChangeIcon->setIcon( icon );
+  }
+  else
+  {
+    checkIcon->setChecked( false );
+    btnChangeIcon->setIcon( QIcon() );
+  }
+  mSymbol = style.symbol();
+  QFont font = style.font();
+  mFontBoldBtn->setChecked( font.bold() );
+  mFontItalicBtn->setChecked( font.italic() );
+  mFontStrikethroughBtn->setChecked( font.strikeOut() );
+  mFontUnderlineBtn->setChecked( font.underline() );
+  mFontFamilyCmbBx->setFont( font );
+}
+
+void QgsFieldConditionalFormatWidget::deleteRule()
+{
+  QgsFieldUIProperties props = mLayer->fieldUIProperties( mFieldCombo->currentField() );
+  QList<QgsConditionalStyle> styles = props.getConditionalStyles();
+  styles.removeAt( mEditIndex );
+  props.setConditionalStyles( styles );
+  mLayer->setFieldUIProperties( mFieldCombo->currentField(), props );
+  pages->setCurrentIndex( 0 );
+  reloadStyles();
+  emit rulesUpdates();
+}
+
+void QgsFieldConditionalFormatWidget::cancelRule()
+{
+  pages->setCurrentIndex( 0 );
+  reloadStyles();
+  reset();
+}
+
+void QgsFieldConditionalFormatWidget::addNewRule()
+{
+  pages->setCurrentIndex( 1 );
+  reset();
+}
+
+void QgsFieldConditionalFormatWidget::reset()
+{
+  mSymbol = 0;
+  mRuleEdit->clear();
+  btnBackgroundColor->setColor( QColor() );
+  btnTextColor->setColor( QColor() );
+  mDefault1->toggle();
+  defaultPressed( mDefault1 );
+  mDeleteButton->hide();
+  mEditing = false;
+  checkIcon->setChecked( false );
+  btnChangeIcon->setIcon( QIcon() );
+
+  mFontBoldBtn->setChecked( false );
+  mFontItalicBtn->setChecked( false );
+  mFontStrikethroughBtn->setChecked( false );
+  mFontUnderlineBtn->setChecked( false );
+}
+
+void QgsFieldConditionalFormatWidget::saveRule()
+{
+  QgsFieldUIProperties props = mLayer->fieldUIProperties( mFieldCombo->currentField() );
+  QList<QgsConditionalStyle> styles = props.getConditionalStyles();
+  QgsConditionalStyle style = QgsConditionalStyle();
+
+  style.setRule( mRuleEdit->text() );
+
+  QColor backColor = btnBackgroundColor->color();
+  QColor fontColor = btnTextColor->color();
+
+  QFont font = mFontFamilyCmbBx->currentFont();
+  font.setBold( mFontBoldBtn->isChecked() );
+  font.setItalic( mFontItalicBtn->isChecked() );
+  font.setStrikeOut( mFontStrikethroughBtn->isChecked() );
+  font.setUnderline( mFontUnderlineBtn->isChecked() );
+  style.setFont( font );
+  style.setBackgroundColor( backColor );
+  style.setTextColor( fontColor );
+  if ( mSymbol && checkIcon->isChecked() )
+  {
+    style.setSymbol( mSymbol->clone() );

```

Comment:
ahh... i see.. it's cloned here. I think it makes more sense to clone it in QgsConditionalStyle rather than here, it's more predictable that way.